### ISSUE TITLE
Use Optionator's new type alias feature

### ISSUE THREAD

**Issue by gkz** [eslint/eslint#4653]
The latest version of [Optionator](https://github.com/gkz/optionator) added a type alias feature. This allows you to add aliases to the types used for parsing and checking input, and displaying help.

I propose that eslint adds type aliases:
```js
{
    Path: "String",
    "[Path]": "[String]"
}
```

Which would allow eslint to use "Path" instead of "String" when appropriate. This would simplify and shorten the help text.

Affected options:
* `--config`, `--cache-file`, `--cache-location`, `--ignore-path`, `--output-file` would change their type from `path::String` to simply `Path`
* `--rulesdir` from `[path::String]` to `[Path]`

**Comment by eslintbot** [comment]
@gkz Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by nzakas** [comment]
I'm not sure "Path" is clearer than "path::String". What do others think?

**Comment by ilyavolodin** [comment]
Is the only difference in the new way of doing it just shorter string? If that's the case, I'm not sure it's worth it.

**Comment by gkz** [comment]
I believe it looks cleaner and it's only a couple of lines of code. But if you don't think it's an improvement, that's fine, that's why I made an issue before creating a pull-request!

In the future I hope to expand on this idea by allowing custom types, so you could set up a custom "Path" type that is not just an alias, but also parsers and checks to make sure it is in the proper format. When I add this functionality maybe we can revisit this.

**Comment by nzakas** [comment]
Again, I think we need to show that there is some benefit to the end user before making any changes. "Cleaner" is really subjective, so I'm not sure we can go by that. In my mind, this is removing "string" from the description, and I'm not sure if that makes it more or less confusing.
